pixeltable/env.py

```diff
@@ -346,16 +354,39 @@ def _set_up(self, echo: bool = False, reinit_db: bool = False) -> None:
 
         self.clear_tmp_dir()
 
-        self._db_name = os.environ.get('PIXELTABLE_DB', 'pixeltable')
-        self._pgdata_dir = Path(os.environ.get('PIXELTABLE_PGDATA', str(Config.get().home / 'pgdata')))
-
-        # cleanup_mode=None will leave the postgres process running after Python exits
-        # cleanup_mode='stop' will terminate the postgres process when Python exits
-        # On Windows, we need cleanup_mode='stop' because child processes are killed automatically when the parent
-        # process (such as Terminal or VSCode) exits, potentially leaving it in an unusable state.
-        cleanup_mode = 'stop' if platform.system() == 'Windows' else None
-        self._db_server = pixeltable_pgserver.get_server(self._pgdata_dir, cleanup_mode=cleanup_mode)
-        self._db_url = self._db_server.get_uri(database=self._db_name, driver='psycopg')
+        db_connect_str = config.get_string_value('DB_CONNECT_STR')
+        if db_connect_str is not None:
+            try:
+                db_url = sql.make_url(db_connect_str)
+                self._db_url = db_url.render_as_string(hide_password=False)
+                self._db_name = db_url.database  # use the dbname given in connect string
+                dialect = db_url.get_dialect().name
+                if dialect == 'cockroachdb':
+                    self._dbms = CockroachDbms(db_url)
+                elif dialect == 'postgresql':
```

Comment:
What is this usage scenario? This is clearly making assumptions that aren't valid in the local case.